onnxruntime/core/optimizer/matmul_transpose_fusion.cc

```diff
@@ -32,14 +32,19 @@ static bool GetTransposePerms(const Node& transpose_node, std::vector<int64_t>&
   return true;
 }
 
-static Node* GetTransposeNodeFromOutput(Graph& graph, NodeArg& node_arg) {
+// is_trans is whether to transpose the 2 dims used to MatMul.
+// is_trans_batch is whether to transpose 1st dim and batch dims (dim-1 to dim-rank-2).
+// For example:
```

Comment:
it would be nice if we can give a more descriptive comments covering what exact cases we target to fuse. 

 An example FYI [https://github.com/microsoft/onnxruntime/blob/91096781c34a5278c496bbd94fd0da4445a3e005/onnxruntime/core/optimizer/bias_softmax_fusion.cc#L85](https://github.com/microsoft/onnxruntime/blob/91096781c34a5278c496bbd94fd0da4445a3e005/onnxruntime/core/optimizer/bias_softmax_fusion.cc#L85)

Reply:
I think it is better to use a different word than "batch" because it is used with respect to training batch. May be something like "range" may be okay.

Reply:
Mathematically speaking this is a cycle of a fixed length less than or equal to the dimensions. (1->0, 2->1,.., r->r-1, 0->r). A transpose is a cycle of length 2.

Reply:
may be "circular permutation" is more clear.

Reply:
CUDA's APIs (GemmBatched, GemmStridedBatched) use the same name. Our MatMul code also calls them batches. I think we still call batch here, but add more comments to explain.